src/Microsoft.Diagnostics.Runtime/src/Implementation/ClrmdModule.cs

```diff
@@ -69,8 +69,8 @@ public ClrmdModule(ClrAppDomain parent, IModuleHelpers helpers, ulong addr)
             {
                 if (_pdb is null)
                 {
-                    using ReadVirtualStream stream = new ReadVirtualStream(_helpers.DataReader, (long)ImageBase, (long)(Size > 0 ? Size : 0x1000));
-                    PEImage pefile = new PEImage(stream, !IsFileLayout);
+                    using ReadVirtualStream stream = new ReadVirtualStream(_helpers.DataReader, (long)ImageBase, (long)(Size > 0 ? Size : int.MaxValue));
+                    using PEImage pefile = new PEImage(stream, !IsFileLayout);
```

Comment:
Unrelated to your change, this should probably use `DataTarget.LoadPEImage`.  Can you make that change here?  Be sure not to dispose it.

Reply:
Also `lock (pefile)` as its used here.

Reply:
> this should probably use `DataTarget.LoadPEImage`

What? `DataTarget.LoadPEImage` only load files on disk.

Reply:
Ah right.  I should probably unify those two paths (or at least refresh my understanding of why these are different), but that's not something for now.